### PR TITLE
Add support for eth/67

### PR DESCRIPTION
**Pull request by marcindsobczak**
!!! DO NOT MERGE IT BEFORE #https://github.com/NethermindEth/nethermind/pull/4475!!!
PR mentioned above is needed for snap sync healing phase after dropping `GetNodeData`

EIP: https://eips.ethereum.org/EIPS/eip-4938

## Changes:
- add support for eth/67 and enable it only if `snap` protocol is enabled

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V67/Eth67ProtocolHandler.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    public override string Name => "eth67";
+
+    public override byte ProtocolVersion => 67;
+
+    public override void HandleMessage(ZeroPacket message)
+    {
+        switch (message.PacketType)
+        {
+            case Eth66MessageCode.GetNodeData:
+                break;
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V67/Eth67ProtocolHandler.cs:57]: So essentially, it will just timeout?

Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V67/Eth67ProtocolHandler.cs:57]: hmm, good point, I will rethink it

Comment (asdacap) [reply src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V67/Eth67ProtocolHandler.cs:57]: What does the spec says?

**Review by asdacap — APPROVED** [review]

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V67/Eth67ProtocolHandler.cs (hunk 1 above)
Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V67/Eth67ProtocolHandler.cs:57]: Spec obligates us to remove message types `GetNodeData` and `NodeData` from the `eth` protocol, is not specifying our exact behaviour:
https://eips.ethereum.org/EIPS/eip-4938#specification
So I think we should do the same as we do when receiving other not supported messages - ignore without respond

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**